frontend/app/components/filters/filter-model/filter-model.service.ts

```diff
@@ -60,12 +60,30 @@ function filterModel(OPERATORS_NOT_REQUIRING_VALUES, SELECTABLE_FILTER_TYPES) {
     },
 
     parseSingleValue: function (v) {
-      return (this.type == 'integer') ? parseInt(v) : v;
+      switch (this.type) {
+        case 'integer':
+          this.textValue = parseInt(v);
+          break;
+        case 'date':
+        case 'date_past':
+          switch (this.operator) {
+            case '=d':
+              this.dateValue = v;
+              break;
+            default:
+              this.textValue = parseInt(v);
+          }
```

Comment:
The problem is that the type information is encoded into the operator, while the filter itself, based on the model attribute simply states ``date`` or ``date_past``, both of which do not convey any type information but merely a behavioral distinction of what filter operators become available when selecting for example created_at / start_date as the attribute to filter.

I fear that in order to get rid of the above, the type information needs to be a part of the operator, e.g. ```'=d`' would become ``{type:'date', op: '='}`` and which would also be the information that is then send to the server along with the values.